Core/DomainModel/EntityCollection.cs

```diff
@@ -127,6 +127,32 @@ public Entity FindEntity(Guid guid)
             return entities[guid];
         }
 
+        public List<Entity> FindEntitiesByTag(string tag)
+        {
+            List<Entity> results = new List<Entity>();
+            lock (entities)
+            {
+                foreach(var entry in entities)
+                {
+                    if (entry.Value.Tags.Contains(tag))
+                    {
+                        results.Add(entry.Value);
+                    }
+                }
+            }
+            return results;
+        }
+
+        public List<Entity> FindEntitiesByTags(List<string> tags)
```

Comment:
Is there a best practice concerning function naming for these cases? I see a possible solution by just overriding the method, and in the case for a single tag, call the generic one with the list by wrapping the single tag in a list tat contains this tag as the only element

Reply:
of course this would be possible by overloading the method. i am just not sure what the functions name should be. i chose the plural s because of a tag list and no plural for a single tag

in hindsight this would even provide a faster solution since we dont have to lookup the entire collection for every tag as i proposed